### PR TITLE
Add HTTP Mixin

### PR DESCRIPTION
**Pull request by schacht-certat**
- Add a HttpMixin
- Use the HttpMixin in the collector bots

### CODE REVIEW

File: intelmq/lib/mixins/http.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            max_retries = self.http_timeout_max_tries - 1
+
+        if self.http_timeout_sec is not None:
+            timeout = self.http_timeout_sec
+
+        adapter = TimeoutHTTPAdapter(max_retries=max_retries, timeout=timeout)
+        self.__session.mount('http://', adapter)
+        self.__session.mount('https://', adapter)
+
+    def http_get(self, url: str) -> requests.Response:
```
Comment (wagner-certat) [inline intelmq/lib/mixins/http.py:100]: If you add kwargs here, the stream collector can pass on the stream flag

File: intelmq/lib/mixins/http.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    http_timeout_max_tries: int = 3
+    http_timeout_sec: int = 30
+    http_username = None
+    http_password = None
+
+    def __init__(self, **kwargs):
+        self.logger.debug("Running HTTP Mixin initialization.")
+        self.http_header = {}
+
+    def setup(self):
```
Comment (wagner-certat) [inline intelmq/lib/mixins/http.py:55]: That almost duplicates the code in `intelmq.lib.utils.create_request_session`

File: intelmq/lib/mixins/http.py (hunk 1 above)
Comment (schacht-certat) [reply intelmq/lib/mixins/http.py]: I've added kwargs and updated the stream collector accordingly

File: intelmq/lib/mixins/http.py (hunk 2 above)
Comment (schacht-certat) [reply intelmq/lib/mixins/http.py:55]: Thats the goal ;)

**Review by wagner-certat — APPROVED** [review]

**Pull request merged**